dandiapi/api/views/common.py

```diff
@@ -58,3 +60,20 @@ class DandiPagination(PageNumberPagination):
     required=True,
     pattern=r'(0\.\d{6}\.\d{4})|draft',
 )
+
+
+def paginated_response(cls, queryset: QuerySet, *args, **kwargs):
+    """
+    Return a paginated response from the provided queryset, using the provided cls.
+
+    Note:
+        * cls should inherit rest_framework.generics.GenericAPIView
```

Comment:
FWIW `should` can be easily expressed in code via 

```python
assert issubclass(cls, GenericAPIView)
```

thus alerting a developer who needs to RTFM without making him reveal to his peers that he didn't bother to RTFM.
I bet that check is fast enough to not worry about impact on performance.